Orange/widgets/data/owsave.py

```diff
@@ -22,58 +39,108 @@ class Inputs:
     resizing_enabled = False
 
     last_dir = Setting("")
-    last_filter = Setting("")
     auto_save = Setting(False)
+    filetype = Setting(FILE_TYPES_NAMES[0])
+    compression = Setting(COMPRESSIONS_NAMES[0])
+    compress = Setting(False)
 
     @classmethod
     def get_writers(cls, sparse):
         return [f for f in FileFormat.formats
                 if getattr(f, 'write_file', None) and getattr(f, "EXTENSIONS", None)
                 and (not sparse or getattr(f, 'SUPPORT_SPARSE_DATA', False))]
 
+    def get_writer_selected(self):
+        type = FILE_TYPES[self.filetype]
+        compression = COMPRESSIONS[self.compression] if self.compress else ''
+        writer = FileFormat.get_reader(type)
+        writer.EXTENSIONS = [writer.EXTENSIONS[writer.EXTENSIONS.index(type + compression)]]
+        return writer
+
+    @classmethod
+    def remove_extensions(cls, filename):
+        if not filename:
+            return None
+        for ext in pathlib.PurePosixPath(filename).suffixes:
+            filename = filename.replace(ext, '')
+        return filename
+
     def __init__(self):
         super().__init__()
         self.data = None
         self.filename = ""
         self.writer = None
 
+        form = QFormLayout(
+            labelAlignment=Qt.AlignLeft,
+            formAlignment=Qt.AlignLeft,
+            rowWrapPolicy=QFormLayout.WrapLongRows,
+            verticalSpacing=10,
+        )
+
+        box = gui.vBox(self.controlArea, "Format")
+
+        gui.comboBox(
+            box, self, "filetype",
+            callback=None,
+            items=FILE_TYPES_NAMES,
+            sendSelectedValue=True,
+        )
+        form.addRow("File type", self.controls.filetype, )
+
+        gui.comboBox(
+            box, self, "compression",
+            callback=None,
```

Comment:
Instead of `callback=None` for both of these combo boxes, you could call a function that fixes self.filename.
E.g. I save a tab file with gzip compression to `test.tab.gz`, then when changing to csv, the stored filename could be updated to `test.csv.gz` and similarly for changes to compression.
Currently, after changing the type and clicking Save again, the file is stored with the previous type since that is what the stored filename reflects...